pkg/clusteragent/externalmetrics/model/datadogmetricinternal.go

```diff
@@ -131,7 +141,23 @@ func NewDatadogMetricInternalFromExternalMetric(id, query, metricName, autoscale
 	}
 }
 
-// Query returns the query that should be used to fetch metrics
+// NewDatadogMetricForTests creates a new internal metric for tests.
```

Comment:
As it's only used in tests, it should directly in the test file

Reply:
I can't because `query` is private from where I need this function